ocp_resources/node_network_configuration_policy.py

```diff
@@ -370,26 +370,37 @@ def _process_failed_status(self, failed_condition_reason):
 
         raise NNCPConfigurationFailed(f"Reason: {failed_condition_reason}\n{last_err_msg}")
 
-    @retry(wait_timeout=120, sleep=5)
     def wait_for_status_success(self):
         failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
         no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
 
-        available_condition = [
-            condition
-            for condition in self.instance.get("status", {}).get("conditions", [])
-            if condition and condition["type"] == self.Conditions.Type.AVAILABLE
-        ]
-        if available_condition:
-            if available_condition[0]["status"] == self.Condition.Status.TRUE:
-                self.logger.info(f"NNCP {self.name} configured Successfully")
-                return available_condition
-
-            elif available_condition[0]["reason"] == no_match_node_condition_reason:
-                raise NNCPConfigurationFailed(f"{self.name}. Reason: {no_match_node_condition_reason}")
-
-            elif available_condition[0]["reason"] == failed_condition_reason:
-                self._process_failed_status(failed_condition_reason=failed_condition_reason)
+        def _get_available_status_condition() -> dict[str, str]:
+            conditions = [
+                condition
+                for condition in self.instance.get("status", {}).get("conditions", [])
+                if condition and condition["type"] == self.Conditions.Type.AVAILABLE
+            ]
+            return conditions[0] if conditions else {}
+
+        try:
+            for sample in TimeoutSampler(
+                wait_timeout=self.success_timeout, sleep=5, func=_get_available_status_condition
```

Comment:
If you cannot use the `retry`, then maybe remove that inner function with:
```python
`func=lambda: next((
            condition
            for condition in self.instance.get("status", {}).get("conditions", [])
            if condition and condition["type"] == self.Conditions.Type.AVAILABLE
), {})
```

Reply:
Yes, this would be cleaner. Let me try it and see how that goes.